mod_app/templates/components/swiper_gallery.html

```diff
@@ -1,22 +1,49 @@
-<div class="swiper">
-  <div class="swiper-wrapper">
-    <!-- Slides -->
-    {% for slide in slides%} {% if slide_images %}
-      <div class="swiper-slide">
-        <img src="{{slide}}" alt="Slide Image" onerror="this.style.display='none'; this.nextElementSibling.style.display='block';" />
-        <div class="fallback-text">Image failed to load: {{slide}}</div>
+{% if not slides %}
+  <div class="fallback-text">
+    <h4>
+      Sorry, no relevant video/images can be found.
+      <h4>
       </div>
-    {% elif slide_videos %}
-      <div class="swiper-slide">
-        <iframe src="{{slide}}" allowfullscreen frameborder="0" scrolling="no" title="Slide Video"></iframe>
-      </div>
-    {% else %}
-      <div class="slide__content">{{slide}}</div>
-    {% endif %} {%endfor %}
+{% elif slides|length > 1 %}
+  <div class="swiper">
+    <div class="swiper-wrapper">
+      {% for slide in slides%}
+        {% if slide_images %}
+          <div class="swiper-slide">
+          <!-- displays fallback-text when image fails to load-->
+            <img class="slide__image" src="{{slide}}" alt={{slide.description| default:'Slide Image'}} onerror="this.style.display='none'; this.nextElementSibling.style.display='block';" />
+            <div class="fallback-text">Slide Image failed to load: {{slide.description|default:slide}}</div>
```

Comment:
since the hover already has the description, I'm not sure there's much merit displaying the description here as well, we could just show the failed to load msg